internal_workflow_testsuite.go

```diff
@@ -343,8 +341,10 @@ func (env *testWorkflowEnvironmentImpl) executeActivity(
 		input,
 	)
 
-	// ensure activityFn is registered to defaultTestTaskList
-	env.testSuite.RegisterActivity(activityFn)
+	if _, ok := getHostEnvironment().getActivityFn(fnName); !ok {
```

Comment:
That's valid, we can remove the Register* methods on the TestSuite. I want to keep the functionality to isolate tests if possible. Having a private env on the testsuite should be simple and not confusing. Self-contained tests enable us to do a lot eg. run tests in parallel.